src/vellum/workflows/nodes/experimental/tool_calling_node/node.py

```diff
@@ -0,0 +1,127 @@
+from collections.abc import Callable
+from typing import ClassVar, List, Optional
+
+from vellum import ChatMessage, FunctionDefinition, PromptBlock
+from vellum.workflows.context import execution_context, get_parent_context
+from vellum.workflows.errors.types import WorkflowErrorCode
+from vellum.workflows.exceptions import NodeException
+from vellum.workflows.graph.graph import Graph
+from vellum.workflows.inputs.base import BaseInputs
+from vellum.workflows.nodes.bases import BaseNode
+from vellum.workflows.nodes.experimental.tool_calling_node.utils import (
+    ToolRouterNode,
+    create_function_node,
+    create_tool_router_node,
+)
+from vellum.workflows.outputs.base import BaseOutputs
+from vellum.workflows.state.base import BaseState
+from vellum.workflows.state.context import WorkflowContext
+from vellum.workflows.types.core import EntityInputsInterface, MergeBehavior
+from vellum.workflows.workflows.base import BaseWorkflow
+
+
+class ToolCallingNode(BaseNode):
+    """
+    A Node that allows for tool calling while dynamically generating a graph based on provided functions.
+
+    Attributes:
+        ml_model: str - The model to use for tool calling (e.g., "gpt-4o-mini")
+        blocks: List[PromptBlock] - The prompt blocks to use (same format as InlinePromptNode)
+        functions: List[FunctionDefinition] - The functions that can be called
+        function_callables: List[Callable] - The callables that can be called
+        prompt_inputs: Optional[EntityInputsInterface] - Mapping of input variable names to values
+    """
+
+    ml_model: ClassVar[str] = "gpt-4o-mini"
+    blocks: ClassVar[List[PromptBlock]] = []
+    functions: ClassVar[List[FunctionDefinition]] = []
+    function_callables: ClassVar[List[Callable]] = []
+    prompt_inputs: ClassVar[Optional[EntityInputsInterface]] = None
+
+    class Trigger(BaseNode.Trigger):
+        merge_behavior = MergeBehavior.AWAIT_ANY
```

Comment:
[nit] can remove - we want all nodes going forward to be AWAIT_ATTRIBUTES by default